extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -81,7 +81,7 @@ static public function onArticleViewHeader( &$article, &$outputDone, &$useParser
 			$app->wg->SuppressPageHeader = true;
 
 			$wallMessage = WallMessage::newFromTitle( $mainTitle );
-			$isDeleted = !$wallMessage->isVisible( $app->wg->User );
+			$isDeleted = $wallMessage->isAdminDelete();
 			$showDeleted = ( $wallMessage->canViewDeletedMessage( $app->wg->User )
 				&& $app->wg->Request->getVal( 'show' ) == '1' );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)